.github/workflows/_ephemeral_environment_manager.yml

```diff
@@ -0,0 +1,117 @@
+        type: string
+        required: true
+      project:
+        type: string
+        default: server
+      cleanup_config:
+        type: boolean
+      sync_environment:
+        type: boolean
+      pull_request_number:
+        type: number
+
+env:
+  _KEY_VAULT: bitwarden-ci
+  _BOT_NAME: bitwarden-devops-bot
+
+jobs:
+  check-run:
+    name: Check PR run
+    uses: ./.github/workflows/check-run.yml
+
+  cleanup:
+    name: Cleanup config
+    if: ${{ inputs.cleanup_config }}
+    runs-on: ubuntu-24.04
+    needs: check-run
+    steps:
+      - name: Login to Azure - Prod Subscription
+        uses: Azure/login@a65d910e8af852a8061c627c456678983e180302 # v2.2.0
+        with:
+          creds: ${{ secrets.AZURE_KV_CI_SERVICE_PRINCIPAL }}
+
+      - name: Retrieve secrets
+        id: retrieve-secrets
+        uses: bitwarden/gh-actions/get-keyvault-secrets@main
+        with:
+          keyvault: ${{ env._KEY_VAULT }}
+          secrets: "github-pat-bitwarden-devops-bot-repo-scope,github-bitwarden-devops-bot-email"
+
+      - name: Checkout ${{ inputs.project }}
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: bitwarden/${{ inputs.project }}
+          ref: ${{ inputs.ephemeral_env_branch }}
+          token: '${{ steps.retrieve-secrets.outputs.github-pat-bitwarden-devops-bot-repo-scope }}'
+
+      - name: Remove config
+        working-directory: ephemeral-environments
+        run: rm -f ${{ inputs.ephemeral_env_branch }}.yaml
+
+      - name: Commit changes to ${{ inputs.ephemeral_env_branch }}
+        working-directory: ephemeral-environments
+        run: |
+          git config --local user.email "${{ steps.retrieve-secrets.outputs.github-bitwarden-devops-bot-email }}"
+          git config --local user.name "${{ env._BOT_NAME }}"
+
+          git add ${{ inputs.ephemeral_env_branch }}.yaml
+          git commit -m "Removed ${{ inputs.ephemeral_env_branch }}.yaml config."
+          git push
+
+  sync-env:
+    name: Sync Ephemeral Environment
+    if: ${{ inputs.sync_environment }}
+    runs-on: ubuntu-24.04
+    needs: check-run
+    steps:
+      - name: Login to Azure - Prod Subscription
+        uses: Azure/login@a65d910e8af852a8061c627c456678983e180302 # v2.2.0
+        with:
+          creds: ${{ secrets.AZURE_KV_CI_SERVICE_PRINCIPAL }}
+
+      - name: Retrieve secrets
+        id: retrieve-secrets
+        uses: bitwarden/gh-actions/get-keyvault-secrets@main
+        with:
+          keyvault: ${{ env._KEY_VAULT }}
+          secrets: "ephemeral-environment-argocd-cluster-url,ephemeral-environment-argocd-cluster-api-secret,ephemeral-environment-argocd-cluster-api-user"
+
+      - name: Install ArgoCD CLI
+        run: |
+          curl -sSL -o argocd-linux-amd64 \
+            "https://github.com/argoproj/argo-cd/releases/latest/download/argocd-linux-amd64"
+
+          install -m 555 argocd-linux-amd64 /usr/local/bin/argocd
```

Comment:
Does this output the version it downloads, or would `argo version` be good to add after it's installed since it's pulling the latest?

Reply:
No it will not, because curl is using the suppressed output, `-s`, flag. Would the version just be for information purposes here?

Reply:
Yeah, just since we're pulling the latest CLI version in case there's ever any hiccups because of a newer released version so we'd be able to see "Build A was on vX and worked, but Build B is on vY and didn't" type of thing.